src/local-development/code-pull-deploy.ts

```diff
@@ -159,6 +170,9 @@ export async function pullComponentCodes(
 	componentMetadata: AppComponentMetadataWithCodeFiles,
 ): Promise<void> {
 	log('debug', `Pull ${appComponentType} ${remoteComponentName}: all codes`);
+
+	sendTelemetry('pull_component_codes', { appComponentType, remoteComponentName });
+
```

Comment:
-- Same as above --

On other places, you have telemetry AFTER real action. Here is is BEFORE. For consistency choose the one variant and folow it everywhere.
If you will call if BEFORE, it will be logged on every actions (includes failed actions). If you will call it AFTER, then the event not logged on function error.

Reply:
moved before